src/main/scala/org/allenai/pipeline/s3/S3Artifact.scala

```diff
@@ -49,6 +53,16 @@ class S3FlatArtifact(
   override def toString: String = s"S3Artifact[${config.bucket}, $path}]"
 }
 
+class CompressedS3FlatArtifact(
+  path: String,
+  config: S3Config,
+  contentTypeOverride: Option[String] = None
+)(implicit cachingImpl: S3Cache = DefaultS3Cache)
+    extends S3FlatArtifact(path, config, contentTypeOverride) with CompressedS3Artifact {
```

Comment:
Could you think through this multiple inheritance mess? ASCII art to the rescue:

```
  CompressedS3FlatArtifact
        |           \
S3FlatArtifact     CompressedS3Artifact
         \          /
          S3Artifact
               |
           Artifact